server.rb

```diff
@@ -9,10 +10,28 @@ def startup
       ['200', {'Content-Type' => 'text/html'}, ["App has started. The time is #{Time.now.strftime("%I:%M:%S %p")}"]]
     end
 
+    Thread.new{UDPEchoServer.new(3030).run}
+
     Rack::Handler::WEBrick.run app, {Host: '0.0.0.0', Daemonize: true}
 
     # TODO create TCP and UDP echo servers and start them
   end
 end
 
+class UDPEchoServer
+  @socket
+
+  def initialize(port)
+    @socket = UDPSocket.new
+    @socket.bind(nil, port)
+  end
+
+  def run
+    while true
+      text, sender = @socket.recvfrom(16)
```

Comment:
16 what, 16 bytes?? We don't want to split up the udp packet into multiple smaller packets, so we should probably use an insanely large buffer